src/util.js

```diff
@@ -47,16 +51,45 @@ export function pathRankSort(a, b) {
 		bAttr = b.attributes || EMPTY;
 	if (aAttr.default) return 1;
 	if (bAttr.default) return -1;
-	let diff = rank(aAttr.path) - rank(bAttr.path);
-	return diff || (aAttr.path.length - bAttr.path.length);
+	let aRank = rank(aAttr.path),
+		bRank = rank(bAttr.path);
+	return (aRank < bRank) ? 1 :
+		(aRank == bRank) ? 0 :
+		-1;
 }
 
 export function segmentize(url) {
 	return strip(url).split('/');
 }
 
-export function rank(url) {
-	return (strip(url).match(/\/+/g) || '').length;
+export function rank(path) {
+	return strip(path).
+	        replace(/(:)?([^\/]*?)([*+?])?(?:\/+|$)/g, (match, isParam, segment, flag) => {
```

Comment:
I was mostly thinking of avoiding the function invocation and replace since it's mutative. That said, since the output is actually being used I'm more amenable to it (now that I properly understand the sort ordering approach).  Only thing I'd change then is to pull out the replace callback and make it a named function.  It's pure, so there's no need to recreate it on each call to `rank()` 👍